src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -225,16 +227,22 @@ bool HasMoreToSync(out BlockHeader[]? headers, out int headersToRequest)
                     {
                         // covering edge case during fastSyncTransition when we're trying to SuggestBlock without the state
                         bool headIsGenesis = _blockTree.Head?.IsGenesis ?? false;
-                        bool toBeProcessedIsNotBlockOne = currentBlock.Number > 1;
-                        bool isFastSyncTransition = headIsGenesis && toBeProcessedIsNotBlockOne;
+                        bool toBeProcessedIsABlockWithNoState = currentBlock.Number > 1;
+                        bool isFastSyncTransition = headIsGenesis && toBeProcessedIsABlockWithNoState;
                         if (isFastSyncTransition)
                         {
-                            long bestFullState = _fullStateFinder.FindBestFullState();
+                            bestFullState ??= _fullStateFinder.FindBestFullState();
```

Comment:
We have `FindBestFullState` 0 even if block that should be processed was already synced, so in case of full sync we request receipts anyway. `bestFullState = currentBlock.Number;` below helped to pass the test, but something else should be done I guess.(Seems like block processing and block downloading is too disconnected)